application/models/TemplateConfiguration.php

```diff
@@ -838,27 +842,46 @@ private function _getImageInfo($file)
         }
         $filePath = $this->_getRelativePath(Yii::app()->getConfig('rootdir'), $file);
         $previewFilePath = App()->getAssetManager()->publish($file);
-        return ['preview' => $previewFilePath, 'filepath' => $filePath, 'filepathOptions' => $filePath ,'filename'=>basename($file)];
+        $fileName = basename($file);
+        return ['preview' => $previewFilePath, 'filepath' => $pathPrefix . $fileName, 'filepathOptions' => $filePath, 'filename' => $fileName];
     }
 
     protected function getOptionPageAttributes()
     {
         $aData = $this->attributes;
-        $fileList = array_merge(Template::getOtherFiles($this->filesPath), Template::getOtherFiles($this->generalFilesPath));
         $aData['maxFileSize'] = getMaximumFileUploadSize();
         $aData['imageFileList'] = [];
-        $categoryList = []; // Array with optgroup label and path
-        $categoryList[] = ['group' => gT("Global"),'path' => $this->generalFilesPath];
-        $categoryList[] = ['group' => gT("Theme"),'path' => $this->filesPath];
-        if($this->sid) {
-            $categoryList[] = ['group' => gT("Survey"),'path' => Yii::app()->getConfig('uploaddir').'/surveys/'.$this->sid.'/images/'];
-        }
-        foreach($categoryList as $category) {
-            $fileList = Template::getOtherFiles($category['path']);
+        $categoryList = $this->getFileCategories();
+
+        // Compose list of image files for each category
+        foreach ($categoryList as $category) {
+            // Get base path for category
+            $pathPrefix = empty($category->pathPrefix) ? '' : $category->pathPrefix;
+            if ($category->name == 'theme') {
```

Comment:
// If the category is theme, add the "files folder" to the base path, as that's the directory to scan for files